src/print-utilities.jl

```diff
@@ -1,11 +1,23 @@
-
 function prettyfloat(f)
     if f == 0
         "0.0"
     elseif f == Inf
         "Inf"
-    elseif (f ≥ 1) && (f - trunc(Int, f) < 1e-8)
-        Printf.@sprintf("%.1f", f)
+    elseif (f ≥ 1)
+        #  ignore "InexactError" if, e.g., f is very large
+        remainder = 1.0
+        try
+            remainder = f - trunc(Int, f)
+        catch e
+            if !isa(e, InexactError)
+                rethrow(e)
+            end
+        end
+        if remainder < 1e-8
+            Printf.@sprintf("%.1f", f)
+        else
+            Printf.@sprintf("%#.5g", f)
+        end
```

Comment:
```suggestion
    elseif ((f ≥ 1) && (f < 1e5) && (f - trunc(Int, f)  < 1e-5))
        Printf.@sprintf("%.1f", f)
```

Since we're printing to 5 sf (dp actually, but because of the scientific notation it's essentially the same thing) in the last branch, lets just check to see if the number has less than 5 digits, and if so, do the slightly nicer printing of the number.